### PR TITLE
Add impellerc options to compile OpenGL Desktop and ES variant shaders.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Also parameterize all unit-tests so all backends are automatically tested.

### CODE REVIEW

File: compiler/includer.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  IncluderData(std::string p_file_name, std::unique_ptr<fml::Mapping> p_mapping)
+      : file_name(std::move(p_file_name)), mapping(std::move(p_mapping)) {}
+};
+
+class Includer final : public shaderc::CompileOptions::IncluderInterface {
+ public:
+  Includer(std::shared_ptr<fml::UniqueFD> working_directory,
+           std::vector<IncludeDir> include_dirs,
+           std::function<void(std::string)> on_file_included);
```
Comment (bdero) [inline compiler/includer.h:29]: This PR made me realize we already support shader `#include`s. Nice!

**Review by bdero — APPROVED** [review]
LGTM

utah teapot

File: compiler/includer.cc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      return mapping;
+    }
+  }
+  return nullptr;
+}
+
+shaderc_include_result* Includer::GetInclude(const char* requested_source,
+                                             shaderc_include_type type,
+                                             const char* requesting_source,
+                                             size_t include_depth) {
```
Comment (zanderso) [inline compiler/includer.cc:72]: Is this used? Is the idea to use it to catch cycles?

**Review by dnfield — APPROVED** [review]

File: compiler/includer.cc (hunk 2 above)
Comment (chinmaygarde) [reply compiler/includer.cc:72]: Do you mean the includer or a specific argument?

File: compiler/includer.h (hunk 1 above)
Comment (chinmaygarde) [reply compiler/includer.h:29]: Right. I remember dragging my feet a bit on whether to wire this up then. But figured it we may end up needing it eventually and didn't want us to hack around not having this.

The includes are also added to the .depfiles generated so editing a header should also recompile any shaders that include that header.

**Comment by chinmaygarde** [comment]
No sure why the ci.yaml validation step is failing and closing/reopening the PR doesn't seem to do the trick anymore. Landing.

File: compiler/includer.cc (hunk 2 above)
Comment (zanderso) [reply compiler/includer.cc:72]: The `include_depth` argument.

What happens if someone messes up such that file A includes file B, and file B includes file A?

**Pull request merged**